encrypt_test.go

```diff
@@ -1358,3 +1369,261 @@ func TestEncrypt(t *testing.T) {
 	}
 	runTestsOverVersions(t, "test", tests)
 }
+
+type secretKeyString string
+
+func newRandomSecretKeyString() (secretKeyString, error) {
+	_, sk, err := box.GenerateKey(rand.Reader)
+	if err != nil {
+		return "", err
+	}
+	return secretKeyString(hex.EncodeToString((*sk)[:])), nil
+}
+
+func (s secretKeyString) toSecretKey() (boxSecretKey, error) {
+	decoded, err := hex.DecodeString(string(s))
+	if err != nil {
+		return boxSecretKey{}, err
+	}
+	private := sliceToByte32(decoded)
+	var public [32]byte
+	curve25519.ScalarBaseMult(&public, &private)
```

Comment:
This is only used for testing, right? I'm slightly scared of this sort of thing in prod code.

Reply:
Yeap, only for testing. Since this is in a `_test.go` file, it's impossible to use it in prod code.